scripts/skills/actives/rf_arrow_to_the_knee_skill.nut

```diff
@@ -121,94 +67,21 @@ this.rf_arrow_to_the_knee_skill <- ::inherit("scripts/skills/skill", {
 		return tooltip;
 	}
 
-	function isUsable()
-	{
-		return this.skill.isUsable() && this.getAmmo() > 0 && !this.getContainer().getActor().isEngagedInMelee();
-	}
-
-	function getAmmo()
-	{
-		local item = this.getContainer().getActor().getItems().getItemAtSlot(::Const.ItemSlot.Ammo);
-
-		if (item == null)
-		{
-			return 0;
-		}
-
-		if (item.getAmmoType() == ::Const.Items.AmmoType.Arrows)
-		{
-			return item.getAmmo();
-		}
-	}
-
-	function consumeAmmo()
-	{
-		local item = this.getContainer().getActor().getItems().getItemAtSlot(::Const.ItemSlot.Ammo);
-
-		if (item != null)
-		{
-			item.consumeAmmo();
-		}
-	}
-
-	function onAfterUpdate( _properties )
-	{
-		this.m.MaxRange = this.m.Item.getRangeMax() + (_properties.IsSpecializedInBows ? 1 : 0);
-		this.m.FatigueCostMult = _properties.IsSpecializedInBows ? ::Const.Combat.WeaponSpecFatigueMult : 1.0;
-	}
-
-	function onUse( _user, _targetTile )
-	{
-		this.consumeAmmo();
-
-		if (!_user.isHiddenToPlayer() || _targetTile.IsVisibleForPlayer)
-		{
-			this.getContainer().setBusy(true);
-			local tag = {
-				Skill = this,
-				User = _user,
-				TargetTile = _targetTile
-				TargetEntity = _targetTile.getEntity()
-			};
-			::Time.scheduleEvent(::TimeUnit.Virtual, this.m.Delay, this.onPerformAttack, tag);
-
-			if (!_user.isPlayerControlled() && _targetTile.getEntity().isPlayerControlled())
-			{
-				_user.getTile().addVisibilityForFaction(::Const.Faction.Player);
-			}
-
-			return true;
-		}
-		else
-		{
-			return this.attackEntity(_user, _targetTile.getEntity());
-		}
-	}
-
 	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
 	{
 		if (_skill == this && _targetEntity.isAlive())
 		{
 			_targetEntity.getSkills().add(::new("scripts/skills/effects/rf_arrow_to_the_knee_debuff_effect"));
+			_targetEntity.getSkills().add(::new("scripts/skills/effects/staggered_effect"));
 		}
 	}
 
-	function onPerformAttack( _tag )
-	{
-		_tag.Skill.getContainer().setBusy(false);
-		local ret = _tag.Skill.attackEntity(_tag.User, _tag.TargetTile.getEntity());
-		return ret;
-	}
-
 	function onAnySkillUsed( _skill, _targetEntity, _properties )
 	{
+		this.quick_shot.onAnySkillUsed(_skill, _targetEntity, _properties);
 		if (_skill == this)
 		{
-			_properties.RangedSkill += this.m.AdditionalAccuracy;
-			_properties.HitChanceAdditionalWithEachTile += this.m.AdditionalHitChance;
-			_properties.RangedDamageMult *= 0.5;
```

Comment:
The description of the skill still says that this attack deals reduced damage but you removed the damage multiplier now. Something is wrong here.

If the damage multiplier is meant to stay then it should be implemented as a member variable.

Reply:
Yeah removal of the damage multiplier (and the defense malus) is intended. I forgot to update the description.